gsva/table_visualization.R

```diff
@@ -0,0 +1,110 @@
+library(reshape2)
+library(ggplot2)
+library(pheatmap)
+
+
+# Rename the columns of a melted table
+rename.melted <- function(input, feature=NA, sample=NA, value=NA, in.place=FALSE) {
+    # Set up the data for use in-or-out-of place
+    if (in.place) {
+        data <- input
+    } else {
+        data <- data.frame(input)
+    }
+    cols <- colnames(data)
+    # Set the feature column
+    if (!is.na(feature)) {
+        if (!(feature %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        feature, ' in data'))
+        }
+        cols[cols==feature] <- 'Feature'
+    }
+    # Set the sample column
+    if (!is.na(sample)) {
+        if (!(sample %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        sample, ' in data'))
+        }
+        cols[cols==feature] <- 'Sample'
+    }
+    # Set the values column
+    if (!is.na(value)) {
+        if (!(value %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        value, ' in data'))
+        }
+        cols[cols==feature] <- 'Value'
+    }
+    # Assign the new column names
+    colnames(data) <- cols
+    return(data)
+}
+
+# Perform all necessary proprocessing steps before visualization
+melt <- function(input, col.as.sample=TRUE) {
```

Comment:
where is this function being used? The name may be problematic when the reshape package also has a melt function

Reply:
Removed, was included for the purpose of future functionality, but will instead be addressed whenever the functionality becomes needed in the future.